### PR TITLE
Add new cops for incorrect platforms / platform families

### PR DESCRIPTION
**Pull request by tas50**
This time let's catch them in case statements. There's some code reuse
in these cops, but it's actually a lot simpler to just write it as two
cops vs. trying to create a unified cop.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: lib/rubocop/cop/chef/correctness/invalid_platform_family_values_in_case.rb [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+                  end
+                end
+              end
+            end
+          end
+
+          def autocorrect(node)
+            new_value = INVALID_PLATFORM_FAMILIES[node.str_content]
+
+            if new_value # some invalid platform families have no direct correction value and return nil instead
```
Comment (phiggins) [inline lib/rubocop/cop/chef/correctness/invalid_platform_family_values_in_case.rb:58]: I'd probably make this a `return unless new_value` and add some whitespace to make it a little more clear this comment is not related to the one below.

File: lib/rubocop/cop/chef/correctness/invalid_platform_values_in_case.rb [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                  end
+                end
+              end
+            end
+          end
+
+          def autocorrect(node)
+            new_value = INVALID_PLATFORMS[node.str_content]
+
+            if new_value # some invalid platform have no direct correction value and return nil instead
```
Comment (phiggins) [inline lib/rubocop/cop/chef/correctness/invalid_platform_values_in_case.rb:58]: Same comment applies here.

Comment (tas50) [reply lib/rubocop/cop/chef/correctness/invalid_platform_values_in_case.rb]: Updated

File: lib/rubocop/cop/chef/correctness/invalid_platform_family_values_in_case.rb (hunk 1 above)
Comment (tas50) [reply lib/rubocop/cop/chef/correctness/invalid_platform_family_values_in_case.rb]: Updated
